### DISCUSSION TITLE
Resolve LINK with Java API

### DISCUSSION THREAD

**Discussion by vic0824** [ArcadeData/arcadedb#1167] [Q&A]
I know it is possible to use the dot notation to resolve a LINK using the SQL API, e.g. `SELECT * FROM Employee WHERE city.name = 'Rome'`.
I was wondering if it is possible to do the same thing using the Java API?

**Comment by sheeehy** [comment]
Not that I know of, but you should be able to retrieve LINKs by accessing properties of documents returned by a query.

**Comment by lvca** [comment]
Once you have a document, you can cross the links with the Java API with something like:

```java
Document employee; // GET THE DOCUMENT IN SOME WAY
String name = ((Identifiable) employee.get("city")).asDocument().getString( "name" );
```

Maybe we could provide a new API to do the job, like:

```java
String name = employee.query("sql", "city.name"));
```

In practice, the `query` method would be similar to the `Database.query()` API but it'd start from the record instead of the database. Under the hood, we could reuse the SQL expressions to execute even more complex stuff. Not sure if it makes sense to provide the language parameter to support the same with Gremlin and other languages.
